generate_table.Rmd

```diff
@@ -0,0 +1,624 @@
+  
+  xml_file <- read_xml(filename)
+  xml_file <- xml_ns_strip(xml_file)
+  
+  # extract each variable; if it isn't present, put NA 
+  extracted <- map(variables, ~{
+    value <- xml_find_all(
+      xml_file, 
+      xpath =.x)
+    value <- ifelse(length(value) ==0, 
+                    NA, 
+                    xml_text(value)) })
+  
+   names(extracted) <- variables_no_path
+   
+   extracted <- extracted %>%
+     as_tibble()
+ 
+ 
+   # add columns with the paths to each variable within the xml file
+   if(include_paths) {
+      paths <- map(variables, ~ {
+        value <- xml_find_all(
+          xml_file, 
+          xpath = paste0(.x)) 
+         value <- ifelse(length(value) ==0, 
+                      NA, 
+                      xml_path(value)) })
+      
+      names(paths) <- paste0(variables_no_path, "_path")
+      
+      extracted <- extracted %>%
+        bind_cols(as_tibble(paths)) 
+  
+   }
+    
+   # check how many of the entries are NA
+   # if all NA, prefix 'irs:' may be needed
+   columns_not_na <- map_dbl(as.data.frame(extracted), ~!is.na(.x)) %>% sum()
+   
+   # handle case where prefix 'irs' is in front
+   if(columns_not_na == 0){
+     
+       extracted <- map(variables_no_path, ~ {
+        value <- xml_find_all(xml_file, 
+                              xpath = paste0("//irs:", .x)) 
+        value <- ifelse(length(value) ==0, 
+                    NA, 
+                    xml_text(value)) })
+        names(extracted) <- variables_no_path
+        
+         extracted <- extracted %>%
+           as_tibble()
+     
+        if(include_paths) {
+        paths <- map(variables_no_path, ~ {
+          value <- xml_find_all(
+            xml_file, 
+            xpath = paste0("//irs:", .x))
+      
+          value <- ifelse(length(value) ==0, 
+                      NA, 
+                      xml_path(value)) })
+        
+        names(paths) <- paste0(variables_no_path, "_path")  
+        
+      extracted <-  extracted %>%
+        bind_cols(as_tibble(paths)) 
+     }
+   }
+   
+ # add name of file to data frame
+ extracted %>%
+   mutate(filename = gsub("*.*\\/", "", filename))
+}
+
+
+
+if(!file.exists("./data")) {dir.create("./data")}
+
+all_data <- map_df(files, ~
+                     get_df(variables = variables_to_extract,
+                     filename = .x)) 
+
+all_data <- all_data %>%
+  mutate(ReturnDate = as.Date(ReturnTs,
+                              format = "%Y-%m-%d")) %>%
+  mutate(across(c(TaxPeriodBeginDt, 
+                  TaxPeriodEndDt),
+                as.Date, 
+                format = "%Y-%m-%d")) %>%
+  mutate(fiscal_year = year(TaxPeriodEndDt),
+         fiscal_year = factor(fiscal_year),
+         BusinessName = toupper(BusinessName))
+
+# check all files are present in data frame
+# nrow(all_data) == length(files)
+
+saveRDS(all_data, "./data/data_990.RDS")
+
+```
+
+
+```{r troubleshooting, include=FALSE,eval=FALSE}
```

Comment:
To confirm, this was your test indicating that you needed to handle the special case of IRS, which you've done, so now the comment in this chunk isn't accurate?